include/aspect/utilities.h

```diff
@@ -838,6 +838,41 @@ namespace aspect
                               const SymmetricTensor<2,dim> &dviscosities_dstrain_rate,
                               const double safety_factor);
 
+
+    /**
+     * This function computes the weighted average $\bar y$ of $y_i$  for a weighted p norm of which
+     * the sum weighs are assumed to be one. This leads for a general p to:
+     * $\bar y = \left(\frac{\sum_{i=1}^k w_i y_i^p}{\sum_{i=1}^k w_i}\right)^{\frac{1}{p}}$.
```

Comment:
out of curiosity, the $l_p$ norm is defined to use `|y_i|^p`, i.e., the p-th power of the *absolute value* of `y_i`. That's because otherwise you get into trouble if some of the values you are averaging are negative. Is there a reason to do it the way you do? What does your implementation do?